### PR TITLE
[Android] Enhancement-561/satelliteCount and satellitesUsedInFix

### PR DESCRIPTION
**Pull request by TimHoogstrate**
Adds satelliteCount and satellitesUsedInFix from the GNSS callback in Android. 

It partly resolves [#561](https://github.com/Baseflow/flutter-geolocator/issues/561).

## Pre-launch Checklist

- [x] I made sure the project builds.
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is does not need version changes.
- [x] I updated `CHANGELOG.md` to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I rebased onto `main`.
- [x] I added new tests to check the change I am making, or this PR does not need tests.
- [x] I made sure all existing and new tests are passing.
- [x] I ran `dart format .` and committed any changes.
- [x] I ran `flutter analyze` and fixed any errors.

[Contributor Guide]: https://github.com/Baseflow/flutter-geolocator/blob/master/CONTRIBUTING.md
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: geolocator_android/android/src/main/java/com/baseflow/geolocator/location/LocationMapper.java [hunk 1]
```diff
@@ -31,10 +31,18 @@ public static Map<String, Object> toHashMap(Location location) {
...
+            position.put("altitude", mslAltitude);
+        }
+        if (location.getExtras().containsKey(NmeaClient.GNSS_SATELLITE_COUNT_EXTRA)) {
+            Double mslSatelliteCount = location.getExtras().getDouble(NmeaClient.GNSS_SATELLITE_COUNT_EXTRA);
+            position.put("gnss_satellite_count", mslSatelliteCount);
+        }
+        if (location.getExtras().containsKey(NmeaClient.GNSS_SATELLITES_USED_IN_FIX_EXTRA)) {
+            Double mslSatellitesUsedInFix = location.getExtras().getDouble(NmeaClient.GNSS_SATELLITES_USED_IN_FIX_EXTRA);
+            position.put("gnss_satellites_used_in_fix", mslSatellitesUsedInFix);
+        }
```
Comment (mvanbeusekom) [inline geolocator_android/android/src/main/java/com/baseflow/geolocator/location/LocationMapper.java:45]: nit: This formatting seems a little bit off, can you format the java code with google-java-format plugin for Android Studio (or IntelliJ)?

File: geolocator_android/android/src/main/java/com/baseflow/geolocator/location/NmeaClient.java [hunk 2]
```diff
@@ -42,6 +49,17 @@ public NmeaClient(@NonNull Context context, @Nullable LocationOptions locationOp
...
           };
+
+        gnssCallback = new GnssStatus.Callback() {
+            @Override
+            public void onSatelliteStatusChanged(@NonNull GnssStatus status) {
+                gnss_satellite_count = status.getSatelliteCount();
+                gnss_satellites_used_in_fix = 0;
+                for (int i = 0; i < gnss_satellite_count; ++i)
+                    if (status.usedInFix(i))
+                        ++gnss_satellites_used_in_fix;
```
Comment (mvanbeusekom) [inline geolocator_android/android/src/main/java/com/baseflow/geolocator/location/NmeaClient.java:60]: Can you use braces here (as per [Google Java Style guide](https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used))?

```suggestion
                for (int i = 0; i < gnss_satellite_count; ++i) {
                    if (status.usedInFix(i)) {
                        ++gnss_satellites_used_in_fix;
                    }
                }
```

File: geolocator_android/lib/src/types/android_position.dart [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+import 'package:meta/meta.dart';
+
+/// Contains additional location information only available on Android platforms.
+@immutable
+class AndroidPosition extends Position {
+  /// Constructs an instance with the given values for testing. [AndroidPosition]
+  /// instances constructed this way won't actually reflect any real information
+  /// from the platform, just whatever was passed in at construction time.
+  const AndroidPosition(
+      {required this.satelliteCount,
```
Comment (mvanbeusekom) [inline geolocator_android/lib/src/types/android_position.dart:12]: nit: small formatting mistake:

```suggestion
  const AndroidPosition({
      required this.satelliteCount,
```

File: geolocator_android/lib/src/types/android_position.dart [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      required timestamp,
+      required accuracy,
+      required altitude,
+      required altitudeAccuracy,
+      required heading,
+      required headingAccuracy,
+      required speed,
+      required speedAccuracy,
+      int? floor,
+      isMocked = false})
```
Comment (mvanbeusekom) [inline geolocator_android/lib/src/types/android_position.dart:25]: nit: add comma after last parameter so `dart format .` will automatically apply correct formatting:

```suggestion
      isMocked = false,
    })
```

File: geolocator_android/lib/src/types/android_position.dart [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  @override
+  String toString() {
+    return 'Latitude: $latitude, Longitude: $longitude, Satellite count: $satelliteCount, Satellites used in fix: $satellitesUsedInFix';
+  }
+
+  @override
+  int get hashCode => satelliteCount.hashCode ^ satellitesUsedInFix.hashCode;
+
+  /// Converts the supplied [Map] to an instance of the [AndroidPosition] class.
+
```
Comment (mvanbeusekom) [inline geolocator_android/lib/src/types/android_position.dart:65]: Remove empty line:
```suggestion
```

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Have some small points, but looks really good.

File: geolocator_android/android/src/main/java/com/baseflow/geolocator/location/LocationMapper.java (hunk 1 above)
Comment (TimHoogstrate) [reply geolocator_android/android/src/main/java/com/baseflow/geolocator/location/LocationMapper.java]: changed the formatting

**Review by mvanbeusekom — APPROVED** [review]
LGTM

**Comment by Wackymax** [comment]
@mvanbeusekom I don't think this was tested at all before merging. The AndroidPosition class zeroes out all the values and the the equals method is flawed checking only the new properties

**Pull request merged**